src/Microsoft.Health.Dicom.Api/Features/Responses/ResourceResult.cs

```diff
@@ -0,0 +1,107 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Health.Dicom.Api.Web;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Messages.Retrieve;
+using Microsoft.Health.Dicom.Core.Web;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.Health.Dicom.Api.Features.Responses
+{
+    internal class ResourceResult : IActionResult
+    {
+        private readonly RetrieveResourceResponse _response;
+        private readonly RetrieveConfiguration _retrieveConfiguration;
+
+        internal ResourceResult(RetrieveResourceResponse response, RetrieveConfiguration retrieveConfiguration)
+        {
+            _response = EnsureArg.IsNotNull(response);
+            _retrieveConfiguration = EnsureArg.IsNotNull(retrieveConfiguration);
```

Comment:
Would it make sense to only pass the property we need off of here?

Reply:
I wanted the resultObject not to know about specific bufferSize, so kept it generic.